error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -889,6 +889,35 @@ Flux<S> after(Flux<T> flux) {
     }
   }
 
+  /** Prefer {@link Flux#mapNotNull(Function)} over more contrived alternatives. */
+  abstract static class FluxMapNotNullMapOptional<T, S> {
+    @Placeholder(allowsIdentity = true)
+    abstract Optional<S> transformation(@MayOptionallyUse T value);
+
+    @BeforeTemplate
+    Flux<S> before(Flux<T> flux) {
+      return flux.map(x -> transformation(x)).mapNotNull(x -> x.orElse(null));
```

Comment:
It turns out that the test for this rule currently fails because `x` is reused. Official Java scoping rules have these variables be independent, but I suspect that Refaster takes a shortcut somewhere. Good that we have tests :)